src/injected/analyzers/rule-analyzer.ts

```diff
@@ -65,6 +69,7 @@ export class RuleAnalyzer extends BaseAnalyzer {
 
     protected onResolve = (analyzerResult: AxeAnalyzerResult): void => {
         this.sendScanCompleteResolveMessage(analyzerResult, this.config);
+        this.postOnResolve(analyzerResult, this.sendMessageDelegate);
```

Comment:
This looks good for the most part but I have a small concern;

It seems odd to me to pipe the send message delegate through this way; ideally, the analyzer shouldn't really know anything about how "postOnResolve" is implemented but this dependency strongly implies that it will be sending a message (and if it isn't, why are we passing the dependency?).

The way to avoid it, in my opinion, would be to make the "sendConvertedResults" into a class object that is given the send message delegate (or you could use bind as well) and then send the object to analyzerProvider as parameter on creation (which can then send it to the specific analyzer).

Reply:
makes sense - updated with a new class whose send function is passed through